### PR TITLE
refactor: Remove @adobe/react-spectrum imports where prohibited

### LINKED ISSUE #2195: Fix linting errors for importing from @adobe/react-spectrum
When completing #1908, we forgot to run the linter on all files locally since CI only lints changed files. As a result, there are some places that violate the rule and fail linting, but weren't caught because the files weren't changed.

I think we should expand the rule to allow `@adobe/react-spectrum` anywhere inside `packages/components`. It's unclear to me what `components/src/spectrum` actually is. We modify and re-export `ComboBox` for example, but then have `SearchableComboBox` in `components/src`. What makes the modified and re-exported `ComboBox` belong in the spectrum folder? I thought they should all be pure re-exports

### PR DESCRIPTION
**Pull request by mattrunyon**
Fixes #2195 

Had to just ignore some of the errors because we compare spectrum components in the styleguide and I didn't want to mess with that for now.

I still think it's a bit confusing that there are things in `components/src/spectrum` that aren't pure re-exports (e.g. ComboBox). Makes it confusing to know which to use.

### CODE REVIEW

File: packages/components/src/dialogs/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
 /* eslint-disable react/jsx-props-no-spreading */
 import { ReactElement, ReactNode } from 'react';
+import type { SpectrumLabelableProps } from '@react-types/shared';
```
Comment (bmingles) [inline packages/components/src/dialogs/ConfirmationDialog.tsx:3]: We may want to include @react-types/shared in the exclusion rule, but I haven't thought through it.

Comment (mattrunyon) [reply packages/components/src/dialogs/ConfirmationDialog.tsx:3]: It creates a circular dependency in some cases. I tried removing it from `@deephaven/react-hooks` and re-exporting from `@deephaven/components`. But `@deephaven/components` already relies on `@deephaven/react-hooks`, so it creates a circular dependency

Comment (bmingles) [reply packages/components/src/dialogs/ConfirmationDialog.tsx:3]: Talked to @mofojed on this. I think this will be fine as-is.

**Review by bmingles — APPROVED** [review]
LGTM. I'd just double check ACL Editor works and nothing weird in things like ConfirmationDialog

**Pull request merged**